### PR TITLE
SLING-8284 - updating parent pom & eliminating dependency on JcrResou…

### PR DESCRIPTION
**Pull request by DominikSuess**
…rceConstants

### CODE REVIEW

**Comment by rombert** [comment]
A SonarQube report for the changes added _only by this pull request_ was generated. Please review it at https://builds.apache.org/job/Sling/job/sling-org-apache-sling-resourcecollection/job/PR-1/1/artifact/target/sonar/issues-report/issues-report-light.html

File: pom.xml [hunk 1]
```diff
@@ -25,8 +25,8 @@
     <parent>
         <groupId>org.apache.sling</groupId>
         <artifactId>sling</artifactId>
-        <version>31</version>
-        <relativePath/>
+        <version>34</version>
+        <relativePath>../../../parent/pom.xml</relativePath>
```
Comment (rombert) [inline pom.xml]: Why did you add this change?

File: src/main/java/org/apache/sling/resource/collection/impl/ResourceCollectionManagerImpl.java [hunk 2]
```diff
@@ -93,14 +92,14 @@ public ResourceCollection createCollection(Resource parentResource, String name,
                 properties = new HashMap<>();
             }
 
-            if (properties.get(JcrResourceConstants.SLING_RESOURCE_TYPE_PROPERTY) != null
-                && !ResourceCollection.RESOURCE_TYPE.equals(properties.get(JcrResourceConstants.SLING_RESOURCE_TYPE_PROPERTY))) {
+            if (properties.get(ResourceResolver.PROPERTY_RESOURCE_TYPE) != null
+                && !ResourceCollection.RESOURCE_TYPE.equals(properties.get(ResourceResolver.PROPERTY_RESOURCE_TYPE))) {
                 properties.put(
-                    JcrResourceConstants.SLING_RESOURCE_SUPER_TYPE_PROPERTY,
+                    "sling:resourceSuperType",
```
Comment (rombert) [inline src/main/java/org/apache/sling/resource/collection/impl/ResourceCollectionManagerImpl.java]: This property seems to be available as a constant, see https://sling.apache.org/apidocs/sling11/org/apache/sling/api/SlingConstants.html#PROPERTY_RESOURCE_SUPER_TYPE . And I think you use this for resource type as well instead of the ResourceResolver property, for the sake of being consistent.

**Review by rombert — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (DominikSuess) [reply pom.xml]: I thought it is good practice to always go with the latest parent - if You're refering to the relative path this is a merge artifact and regression as my initial patch was against an older version so could be dropped.

File: src/main/java/org/apache/sling/resource/collection/impl/ResourceCollectionManagerImpl.java (hunk 2 above)
Comment (DominikSuess) [reply src/main/java/org/apache/sling/resource/collection/impl/ResourceCollectionManagerImpl.java]: @rombert  great - somehow missed the SlingConstants - thx for pointing out

**Comment by rombert** [comment]
A SonarQube report for the changes added _only by this pull request_ was generated. Please review it at https://builds.apache.org/job/Sling/job/sling-org-apache-sling-resourcecollection/job/PR-1/2/artifact/target/sonar/issues-report/issues-report-light.html

**Review by rombert — APPROVED** [review]
LGTM, thanks @DominikSuess !

File: pom.xml (hunk 1 above)
Comment (rombert) [reply pom.xml]: Yes, I mean the relativePath. Thanks for dropping.
